plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/RaptorConnectorFactory.java

```diff
@@ -66,29 +69,31 @@ public Connector create(String catalogName, Map<String, String> config, Connecto
     {
         checkSpiVersion(context, this);
 
-        Bootstrap app = new Bootstrap(
-                new CatalogNameModule(catalogName),
-                new JsonModule(),
-                new MBeanModule(),
-                new ConnectorObjectNameGeneratorModule("io.trino.plugin.raptor.legacy", "trino.plugin.raptor.legacy"),
-                new MBeanServerModule(),
-                binder -> {
-                    binder.bind(NodeManager.class).toInstance(context.getNodeManager());
-                    binder.bind(PageSorter.class).toInstance(context.getPageSorter());
-                    binder.bind(TypeManager.class).toInstance(context.getTypeManager());
-                    binder.bind(CatalogName.class).toInstance(new CatalogName(catalogName));
-                },
-                metadataModule,
-                new BackupModule(backupProviders),
-                new StorageModule(),
-                new RaptorModule(),
-                new RaptorSecurityModule());
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
```

Comment:
No need to pass in the class loader. We can simply do `getClass().getClassLoader()` here.

However, I don't understand why this change is needed at all, since `ConnectorFactory.create()` is already called with the correct class loader: https://github.com/trinodb/trino/blob/1ca635e31eb2215e064e054f2aef38053b8e5426/core/trino-main/src/main/java/io/trino/connector/DefaultCatalogFactory.java#L202-L204

Reply:
I see the problem now. We should fix this in the instrumentation to preserve the correct class loader. We provide this undocumented guarantee for other plugin services. This type of behavior change is likely to break other connectors and is difficult for people to debug.

Reply:
Fixed.